stratum/hal/lib/p4/BUILD

```diff
@@ -143,6 +143,7 @@ stratum_cc_library(
         "//stratum/glue/status",
         "//stratum/glue/status:status_macros",
         "//stratum/glue/status:statusor",
+        "//stratum/hal/lib/tdi:tdi_constants",	# fix this
```

Comment:
The reason behind the comment fixthis ?

Reply:
Because this dependency violates the Stratum architecture.

The hal/lib/p4 modules are _platform neutral_. They do not depend on files in hal/lib/barefoot, hal/lib/bcm, **or hal/lib/tdi**.

We need either to abstract out the dependency or (more likely) to make the constants on which the P4 layer now depends a part of the p4 layer.